File: typescript/src/base_to_base.ts
import { iterate as baseToDec } from "./base_to_dec.ts";
import { iterate as decToBase } from "./dec_to_base.ts";
export const baseToBase: (
  value: string,
  base: number,
  newBase: number,
) => string = (value, base, newBase) => {
  return decToBase(baseToDec(value, base), newBase);
};

File: elixir/lib/polyglot/base_to_base.ex
defmodule Polyglot.BaseToBase do
  alias Polyglot.DecToBase
  alias Polyglot.BaseToDec

  def base_to_base(value, base, new_base) do
    DecToBase.pattern_match(BaseToDec.pattern_match(value, base), new_base)
  end
end
